packages/react-rest/src/services/useAll/useAll.ts

```diff
@@ -58,16 +59,25 @@ export const useAll = <
   const [error, setError] = useState<MetaError | undefined>(undefined)
   const [loading, setLoading] = useState<boolean>(false)
 
-  const fetchAll = useCallback(() => {
+  const wait = (minimumLoadTime: number): Promise<void> => {
+    return new Promise((resolve) => {
+      setTimeout(resolve, minimumLoadTime)
+    })
+  }
+
+  const fetchAll = useCallback(async () => {
     setLoading(true)
-    findAll<TSchemas, TSchemaName>(
-      dataSource,
-      allSchemas,
-      schemaName,
-      memoizedQuery,
-      memoizedBaseFilter,
-    )
-      .then(([data, requestMeta]) => {
+    Promise.all([
+      findAll<TSchemas, TSchemaName>(
+        dataSource,
+        allSchemas,
+        schemaName,
+        memoizedQuery,
+        memoizedBaseFilter,
+      ),
+      wait(minimumLoadTime ? minimumLoadTime : 250),
```

Comment:
we should default to zero here or completely skip it if it's undefined

```
wait(minimumLoadTime ?? 0)
```
or
```
minimumLoadTime ? wait(minimumLoadTime) : undefined
```
not sure if there is a performance difference between those 2, so either _should_ be fine (i think the 2nd one works 😅, but either way default should always be immediately transition away from loading state